source/common/http/conn_manager_impl.cc

```diff
@@ -58,12 +58,18 @@ ConnectionManagerImpl::ConnectionManagerImpl(ConnectionManagerConfig& config,
                                              Runtime::RandomGenerator& random_generator,
                                              Tracing::HttpTracer& tracer, Runtime::Loader& runtime,
                                              const LocalInfo::LocalInfo& local_info,
-                                             Upstream::ClusterManager& cluster_manager)
+                                             Upstream::ClusterManager& cluster_manager,
+                                             Server::OverloadManager* overload_manager)
     : config_(config), stats_(config_.stats()),
       conn_length_(new Stats::Timespan(stats_.named_.downstream_cx_length_ms_)),
       drain_close_(drain_close), random_generator_(random_generator), tracer_(tracer),
       runtime_(runtime), local_info_(local_info), cluster_manager_(cluster_manager),
-      listener_stats_(config_.listenerStats()) {}
+      listener_stats_(config_.listenerStats()),
+      overload_inactive_(Server::OverloadActionState::Inactive),
```

Comment:
TIOLI: provide a static method that returns reference to this value via CONSTRUCT_ON_FIRST_USE instead of using a field